### PR TITLE
Allow null value for access checking if user is guest

### PR DESCRIPTION
**Pull request by rustamwin**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ✔️/❌
| Fixed issues  | #76 

:warning: Merge after https://github.com/yiisoft/access/pull/13

### CODE REVIEW

File: src/Manager.php [hunk 1]
```diff
@@ -49,8 +51,14 @@ public function __construct(
...
     public function userHasPermission($userId, string $permissionName, array $parameters = []): bool
     {
+        if ($userId === null && $this->guestRole !== null) {
+            return $this->guestHasPermission($permissionName);
+        }
+
         if (!is_string($userId) && !is_int($userId)) {
-            throw new InvalidArgumentException(sprintf('userId must be a string or an int, %s given.', gettype($userId)));
+            throw new InvalidArgumentException(
+                sprintf('userId must be a string or an int, %s given.', gettype($userId))
```
Comment (vjik) [inline src/Manager.php:60]: It should be taken into account that `$userId` can be an object:

```php
is_object($userId) ? get_class($userId) : gettype($userId);
```

File: src/Manager.php [hunk 2]
```diff
@@ -474,6 +482,20 @@ public function getDefaultRoleInstances(): array
...
     }
 
+    /**
+     * Set guest role.
+     *
+     * @param string $guestRole
+     *
+     * @return $this
+     */
+    public function setGuestRole(string $guestRole): self
```
Comment (vjik) [inline src/Manager.php:492]:
```suggestion
    public function setGuestRole(?string $guestRole): self
```

File: src/Manager.php [hunk 3]
```diff
@@ -664,6 +686,20 @@ private function userHasPermissionRecursive(
         return false;
     }
 
+    private function guestHasPermission(string $permissionName): bool
+    {
+        /** @psalm-suppress PossiblyNullArgument */
```
Comment (vjik) [inline src/Manager.php:691]:
```suggestion
```

Instead psalm suppress better create private method with annotation `@psalm-assert string $this->guestRole` and call him in `userHasPermission()` method.

File: src/Manager.php [hunk 4]
```diff
@@ -49,8 +51,14 @@ public function __construct(
 
     public function userHasPermission($userId, string $permissionName, array $parameters = []): bool
     {
+        if ($userId === null && $this->guestRole !== null) {
```
Comment (vjik) [inline src/Manager.php:54]: In case when both user ID and guest role is null in this method should return false instead of throws exception.

**Review by samdark — APPROVED** [review]

**Review by samdark — APPROVED** [review]
Looks 👍 Need to mention it in README.

**Review by vjik — APPROVED** [review]

**Pull request merged**